src/components/app/app.jsx

```diff
@@ -1,22 +1,113 @@
-import React from 'react';
+import React, {PureComponent} from 'react';
 import PropTypes from 'prop-types';
+import {BrowserRouter, Route, Switch} from 'react-router-dom';
 import WelcomeScreen from './../welcome-screen/welcome-screen.jsx';
+import QuestionGenreScreen from '../question-genre-screen/question-genre-screen.jsx';
+import QuestionArtistScreen from '../question-artist-screen/question-artist-screen.jsx';
+import {GameType} from "../../const.js";
 
-const welcomeButtonHandler = () => {};
+class App extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.handleWelcomeButtonClick = this.handleWelcomeButtonClick.bind(this);
+    this.handleAnswerSubmit = this.handleAnswerSubmit.bind(this);
+    this.state = {
+      step: -1
+    };
+  }
```

Comment:
```suggestion
  constructor(props) {
    super(props);
    
    this.state = {
      step: -1
    };
    
    this.handleWelcomeButtonClick = this.handleWelcomeButtonClick.bind(this);
    this.handleAnswerSubmit = this.handleAnswerSubmit.bind(this);
  }
```
вот такая очередность больше прижилась в реальных проектах :)

Reply:
ок, учту